reporthandling/results/v1/reportsummary/summarydetails.go

```diff
@@ -57,6 +58,60 @@ func (summaryDetails *SummaryDetails) InitResourcesSummary(controlInfoMap map[st
 	summaryDetails.CalculateStatus()
 }
 
+// initResourcesSummary must run this AFTER initializing the controls
+func (summaryDetails *SummaryDetails) InitSubsectionsSummary(opaFrameworks []reporthandling.Framework, controlInfoMap map[string]apis.StatusInfo) {
```

Comment:
Nested in the FW results you can find the list of controls, so why are you passing it as it's own? You can use what is in the FWs

Reply:
This is the problem I raised in our prior conversation. We indeed have the controls in the framework result, but we don't have all the information about the framework itself except for its name. So we don't have here the subsection info we need to create the subsections summaries. This is why we need to pass the frameworks themselves (`opaFrameworks`), and the controls results (`controlInfoMap`)